File: src/my_project_typescript/easy_problems_typescript/from101to150/twoFurthestHousesWithDifferentColors.ts
function maxDistance(colors: number[]): number {
    let max: number = -1
    let lenColors: number = colors.length

    for (let i = 0; i < lenColors; i++){
        for (let j = i+1; j < lenColors; j++){
            if (Math.abs(colors[i]-colors[j]) !== 0){
                if (j-i > max){
                    max = j-i
                }
            }
        }
    }
    return max     
};

console.log(maxDistance([1,1,1,6,1,1,1]))

console.log(maxDistance([1,8,3,8,3]))

console.log(maxDistance([0,1]))

File: src/my_project/easy_problems/from201to250/two_furthest_houses_with_different_colors.py
from typing import List 

class Solution:
    def maxDistance(self, colors: List[int]) -> int:
        max = -1
        len_colors = len(colors)
        for i in range(len_colors):
            for j in range(i+1, len_colors):
                if abs(colors[i]-colors[j]) != 0:
                    if j-i> max:
                        max = j-i
        
        return max 
    
    
solution = Solution()

print(solution.maxDistance(colors = [1,1,1,6,1,1,1]))

print(solution.maxDistance(colors = [1,8,3,8,3]))

print(solution.maxDistance(colors = [0,1]))
